acceptance: add traffic to rebalancing tests

vivekmenezes opened the pull request:
With lease holder rebalancing working, it makes sense
to add some traffic to the rebalancing tests so that they
are more realistic.

fixes #10465

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/12177)
<!-- Reviewable:end -->

vivekmenezes commented on pkg/acceptance/terraform/examples.tf:
```diff
@@ -0,0 +1,72 @@
+# Various examples that can be run against a cockroach cluster in GCE.
```
this file has been mostly copied from the one in cloud/gce/...

cuongdo commented:
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[pkg/acceptance/terraform/examples.tf, line 1 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-KYUAvHAP4ru9hl9pHO_-r1-1:-KYUBTMsp0Xyg33qb66K:b-uh6efi) ([raw file](https://github.com/cockroachdb/cockroach/blob/df2182568529941af6b2a2ecac29d698189de380/pkg/acceptance/terraform/examples.tf#L1)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary><blockquote>

this file has been mostly copied from the one in cloud/gce/...
</blockquote></details>

Why is this needed? The nightly `block_writer` and `photos` tests run without this file.  See:

https://github.com/cockroachdb/cockroach/blob/master/pkg/acceptance/continuous_load_test.go#L88

Every `cockroach` node already has `photos` and `block_writer`. See:

https://github.com/cockroachdb/cockroach/blob/master/pkg/acceptance/terraform/main.tf#L127

---

*[pkg/acceptance/terraform/main.tf, line 60 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-KYUBtx0ScGlEp3QwcA-:-KYUBtx1UGknt-iMyZEn:b-ua9jji) ([raw file](https://github.com/cockroachdb/cockroach/blob/df2182568529941af6b2a2ecac29d698189de380/pkg/acceptance/terraform/main.tf#L60)):*
> ```HCL
>     # shifted by one (first element is empty), then take the value at index "instance.index".
>     join_address = "${element(concat(split(",", ""), google_compute_instance.cockroach.*.network_interface.0.access_config.0.assigned_nat_ip), count.index)}"
>     # We need to provide one node address for the block writer.
> ```

We don't need a different host for `block_writer`. It uses next to no CPU, so it can run on the first CockroachDB node.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177)*
<!-- Sent from Reviewable.io -->

cuongdo commented:
Also, have you thought about whether the `TESTTIMEOUT` for the `make test` command lines in TeamCity need to be modified to allow more time for the rebalancing to occur during our nightly tests?

vivekmenezes commented:
@cuongdo thanks for the review. I've updated the commit with your comments. I think I'd actually like to follow up this work with another PR that removes the continuous load test and incorporates it along with all its checks completely into this test, and eventually rename the allocator test to be the continuous load test.

vivekmenezes commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/acceptance/terraform/examples.tf, line 1 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-KYUAvHAP4ru9hl9pHO_-r1-1:-KYZwAEFlkmjfPEvsSUu:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/df2182568529941af6b2a2ecac29d698189de380/pkg/acceptance/terraform/examples.tf#L1)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

Why is this needed? The nightly `block_writer` and `photos` tests run without this file.  See:

https://github.com/cockroachdb/cockroach/blob/master/pkg/acceptance/continuous_load_test.go#L88

Every `cockroach` node already has `photos` and `block_writer`. See:

https://github.com/cockroachdb/cockroach/blob/master/pkg/acceptance/terraform/main.tf#L127
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177)*
<!-- Sent from Reviewable.io -->

cuongdo commented:
I wouldn't recommend that. The allocator tests start with existing test data. The continuous load test starts with an empty cluster. There is value in testing both scenarios under load. cc @petermattis

---

Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/acceptance/allocator_test.go, line 285 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-KY_5OGVbCg-Ddfc6YWv:-KY_5OGVbCg-Ddfc6YWw:bn9z8z7) ([raw file](https://github.com/cockroachdb/cockroach/blob/e3748c61d1373900c5b642e00f6f81464865172f/pkg/acceptance/allocator_test.go#L285)):*
> ```Go
> 	return util.RetryForDuration(10*time.Second, func() error {
> 		for i := 0; i < *flagCLTWriters; i++ {
> 			if !started[i] {
> ```

where is `started[i]` set?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-:-KY_5Rag4q2njDOMEEhn:bnczlac)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[pkg/acceptance/allocator_test.go, line 280 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-KY_8rdkVeK0N9rTeI1l:-KY_8rdkVeK0N9rTeI1m:b-ab509u) ([raw file](https://github.com/cockroachdb/cockroach/blob/e3748c61d1373900c5b642e00f6f81464865172f/pkg/acceptance/allocator_test.go#L280)):*
> ```Go
> 	}
> 
> 	// We may have to retry restarting the load generators, because CockroachDB
> ```

If using supervisor, it's possible to have them autorestart and start when the supervisor service loads.  Not sure if that's what you want.

---

*[pkg/acceptance/terraform/main.tf, line 60 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-KYUBtx0ScGlEp3QwcA-:-KY_6ooKjwPWn4HZ8YLi:b-kfls8g) ([raw file](https://github.com/cockroachdb/cockroach/blob/df2182568529941af6b2a2ecac29d698189de380/pkg/acceptance/terraform/main.tf#L60)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

We don't need a different host for `block_writer`. It uses next to no CPU, so it can run on the first CockroachDB node.
</blockquote></details>

We tend to run blockwriter on all nodes.  Are we not going to do that here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177)*
<!-- Sent from Reviewable.io -->

petermattis commented:
> The allocator tests start with existing test data. The continuous load test starts with an empty cluster. There is value in testing both scenarios under load.

Agreed. It's good to test empty clusters due to the splitting that occurs.

vivekmenezes commented:
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[pkg/acceptance/allocator_test.go, line 280 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-KY_8rdkVeK0N9rTeI1l:-KY_EW3sVG6jhKDsXgW9:bezajjq) ([raw file](https://github.com/cockroachdb/cockroach/blob/e3748c61d1373900c5b642e00f6f81464865172f/pkg/acceptance/allocator_test.go#L280)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

If using supervisor, it's possible to have them autorestart and start when the supervisor service loads.  Not sure if that's what you want.
</blockquote></details>

I don't think so

---

*[pkg/acceptance/allocator_test.go, line 285 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-KY_5OGVbCg-Ddfc6YWv:-KY_EIS_xXnzvgN7fOER:bja4uge) ([raw file](https://github.com/cockroachdb/cockroach/blob/e3748c61d1373900c5b642e00f6f81464865172f/pkg/acceptance/allocator_test.go#L285)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

where is `started[i]` set?
</blockquote></details>

hmmm looks like a bug in the original code that I copied :-)

---

*[pkg/acceptance/terraform/main.tf, line 60 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-KYUBtx0ScGlEp3QwcA-:-KY_E8hYCp2bh2_yO8lj:b-1qh3mi) ([raw file](https://github.com/cockroachdb/cockroach/blob/df2182568529941af6b2a2ecac29d698189de380/pkg/acceptance/terraform/main.tf#L60)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

We tend to run blockwriter on all nodes.  Are we not going to do that here?
</blockquote></details>

We're only going to be running 3 (default) block writers here 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
ready for another review. Thanks

vivekmenezes committed:
acceptance: add traffic to rebalancing tests

With lease holder rebalancing working, it makes sense
to add some traffic to the rebalancing tests so that they
are more realistic.

fixes #10465

cuongdo commented:
How long do `TestUpreplicate_1To3Small` and `TestRebalance_3To5Small` take to finish with these changes?

---

Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-:-KYnuN9nWCS5WF4bhPLO:b-67ba7e)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
--- PASS: TestRebalance_3To5Small (1008.53s)
PASS
ok  	github.com/cockroachdb/cockroach/pkg/acceptance	1008.567s

real	17m2.554s
user	0m26.294s
sys	0m3.809s

and

--- PASS: TestUpreplicate_1To3Small (2116.26s)
PASS
ok  	github.com/cockroachdb/cockroach/pkg/acceptance	2116.293s

real	35m20.226s
user	0m9.008s
sys	0m2.989s

---

Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-:-KYohMhE3hhZDvV5to3H:b-9gatiz)*
<!-- Sent from Reviewable.io -->

cuongdo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12177#-:-KYorIW4WfvgWkJf4BDM:bnfp4nl)*
<!-- Sent from Reviewable.io -->
